### ISSUE TITLE
Enable to load data on a specific location

### ISSUE THREAD

**Issue by yahnF** [streetcomplete/StreetComplete#5181]
I am not able to load quests from a specific location in my city. 
It is not a network problem as the error message indicates because even remotly, the bug persists.
I suppose it is due to an incorrect data in OSM, but I did not find anything on Osmose.
This is a long running problem, at least one year. 
PS: I'm french so it could be an accent in an attribute wrongly handled.

**How to Reproduce**

Fetch quests near 43.56815,1.41534
The error message (in french): Network trouble
![error](https://github.com/streetcomplete/StreetComplete/assets/20631460/07ff88c2-5eea-47b1-b496-500e65d0a42a)
The tile in question:
![tile](https://github.com/streetcomplete/StreetComplete/assets/20631460/89f8ce36-d2ec-4a31-b8f6-396f7dbf39d2)

**Expected Behavior**

Quests are loaded

**Versions affected**
v53.3 and earlier

**Comment by Helium314** [comment]
There is some element with a value exceeding the length limit ("limited to less than 256 characters"). The log only shows the value and invalid length error, but not the element.

**Comment by matkoniecz** [comment]
link: https://www.openstreetmap.org/#map=15/43.5682/1.4153

**Comment by matkoniecz** [comment]
problem happens for https://www.openstreetmap.org/node/9671207497 - I bet that it is emoji fault

**Comment by yahnF** [comment]
@matkoniecz was right, it was the emoji.
Removing it and keeping a description longer than 256 chars worked (259).

Do you think it could be possible to support emojis to remove some blind spots?

**Comment by matkoniecz** [comment]
> Do you think it could be possible to support emojis to remove some blind spots?

Yes, that is why this issue will be open until it will be fixed

> Removing it and keeping a description longer than 256 chars worked (259).

Are you sure that it is not shorter than 256? https://www.openstreetmap.org/node/9671207497 `description` has now 254 characters

**Comment by yahnF** [comment]
My mistake, it is 259 octets but 254 chars...

**Comment by westnordost** [comment]
Hm, the error is here, in the osmapi library:

```java
if(value.length() >= 256)
```

https://github.com/westnordost/osmapi/blob/master/libs/map/src/main/java/de/westnordost/osmapi/map/data/OsmTags.java#L40

---

[Java documentation for `String.length()`](https://docs.oracle.com/javase/8/docs/api/java/lang/String.html#length--):

> Returns the length of this string. The length is equal to the number of [Unicode code units](https://docs.oracle.com/javase/8/docs/api/java/lang/Character.html#unicode) in the string.

---

```
"Agence de communication 360 : nous accompagnons tout professionnel dans le domaine du web, mobile, marketing, signalétique et flocage de tous supports pour une visibilité accrue. Développez votre business en boostant votre image de marque 🚀Équipe d'expert".length()
```

returns 256, so the issue has nothing to do with emojis. It's simply that the check in the  `osmapi` library is wrong - a key and value can have **up to** 256 characters.

**Comment by mmd-osm** [comment]
> a key and value can have up to 256 characters.

No, that’s not correct, the API limit is 255 Unicode characters, not 256.

**Comment by westnordost** [comment]
Uf, that's complicates things... I'll have to look into how to get the unicode character count in Java...

**Comment by westnordost** [comment]
Alright, so indeed since Java 8, it is possible to count the unicode code points, with `myString.codePoints().count()`.

**Comment by riQQ** [comment]
Fixed with https://github.com/westnordost/osmapi/commit/78f837044e01ddcaa70bc6c1c4b677b960249309

**Comment by westnordost** [comment]
Oh no... I trusted that Java 8 would now finally be fully supported (via desugar) in Android. Thank you very much for spotting this!

I guess, in this case I should just completely remove this check? It will then fail on upload (if the string is too long) rather than when the string is set.
